src/routes/event/list.tsx

```diff
@@ -16,18 +18,45 @@ export default function EventList() {
   const navigate = useNavigate();
 
   const { allEvents, getAllEvents } = useEventState();
-  console.log("allEvents: ", allEvents);
   const { user } = useAuthState();
 
+  if (!user?.accountType) return null;
+
```

Comment:
you can't have a return null and a useEffect later
your IDE and the browser console should have warned you about it.
You didnt follow those comments i wrote in terms of component structure.